saleor/graphql/checkout/mutations/utils.py

```diff
@@ -98,6 +98,26 @@ def update_checkout_shipping_method_if_invalid(
         clear_delivery_method(checkout_info)
 
 
+def get_variants_and_total_quantities(
+    variants, lines_data, quantity_to_update_check=False
+):
+    variants_total_quantity_map = defaultdict(int)
+    mapped_data = defaultdict(int)
+
+    if quantity_to_update_check:
+        lines_data = filter(lambda d: d.quantity_to_update, lines_data)
+
+    for data in lines_data:
+        mapped_data[data.variant_id] += data.quantity
+
+    for variant in variants:
+        quantity = mapped_data.get(str(variant.id), None)
```

Comment:
You don't need to do that. `defaultdict(int)` returns 0 if key was not present.

Reply:
Oh, OK - you want to omit variants with no quantity from the final list. That works then :+1: